Arrow keys handling

szymonkups opened the pull request:
Fixes #15. Please review after closing: https://github.com/ckeditor/ckeditor5-engine/pull/717.

szymonkups commented on src/widget/widget.js:
```diff
@@ -140,4 +187,69 @@ export default class Widget extends Plugin {
+//
+// @param {module:engine/model/selection~Selection} modelSelection
+// @returns {module:engine/model/element~Element|null}
+function getSelectedElement( modelSelection ) {
```
I was wondering whether this should stay here or be moved as model's selection method. We have similar one in view's selection.

Reinmar replied:
Makes sense moving it there. But it needs to have the same semantics like the method in the view. Are they identical?

szymonkups replied:
Yes, they have same semantics. I will create an issue in [ckeditor5-engine](https://github.com/ckeditor/ckeditor5-engine).

Reinmar commented:
All works except this case:
![dec-13-2016 17-33-06](https://cloud.githubusercontent.com/assets/156149/21149060/790b7e1c-c15a-11e6-817c-3729a97440b2.gif)

szymonkups commented:
I've fixed issue from previous comment and added test checking it. But I noticed one thing - when data is loaded and image is the first element in the document filler is added before it. This is probably an engine issue, could you confirm?

Reinmar commented:
> I've fixed issue from previous comment and added test checking it. But I noticed one thing - when data is loaded and image is the first element in the document filler is added before it. This is probably an engine issue, could you confirm?

Hm... Hard to tell. The `_getDefaltRange()` method is called couple of times when content is being loaded. It's last calls seem to return a range containing the image:

![image](https://cloud.githubusercontent.com/assets/156149/21178412/bf2d928c-c1ef-11e6-86a2-1f9d9de0c98c.png)

So, I think that this is a bug in the engine. I'll report a ticket for it, but please change this sample so image isn't the first element in the root.

Reinmar commented:
Reported https://github.com/ckeditor/ckeditor5-engine/issues/730.

Reinmar commented on src/widget/widget.js:
```diff
@@ -128,6 +130,51 @@ export default class Widget extends Plugin {
+		const modelDocument = this.editor.document;
+		const schema = modelDocument.schema;
+		const modelSelection = modelDocument.selection;
+		const objectElement = getSelectedElement( modelSelection );
```
Should use the new engine function

Reinmar commented on src/widget/widget.js:
```diff
@@ -128,6 +130,51 @@ export default class Widget extends Plugin {
+		const objectElement = getSelectedElement( modelSelection );
+
+		// if object element is selected.
+		if ( objectElement && schema.objects.has( objectElement.name ) ) {
```
This code has nothing to do with widgets. This is interesting, cause it means that we can move all that to the engine easily. That can be a good option. Perhaps it will be easily doable to have basic widget constructs in the engine. However, as a what? Addition to the editing controller? Maaaybe. But it's so much unnecessary code that I don't know ;|.

szymonkups replied:
That's a valid point. We can make a discussion about it when we will extracting widgets functionalities from image plugin.
